src/Microsoft.NetCore.Analyzers/Core/Performance/UsePropertyInsteadOfCountMethodWhenAvailable.cs

```diff
@@ -0,0 +1,282 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UsePropertyInsteadOfCountMethodWhenAvailableMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UsePropertyInsteadOfCountMethodWhenAvailableDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Performance,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultForVsixAndNuget,
+            description: s_localizableDescription,
+#pragma warning disable CA1308 // Normalize strings to uppercase
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/" + RuleId.ToLowerInvariant());
+#pragma warning restore CA1308 // Normalize strings to uppercase
+
+        /// <summary>
+        /// Returns a set of descriptors for the diagnostics that this analyzer is capable of producing.
+        /// </summary>
+        /// <value>The supported diagnostics.</value>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_rule);
+
+        /// <summary>
+        /// Called once at session start to register actions in the analysis context.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        /// <summary>
+        /// Called on compilation start.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        private static void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            if (WellKnownTypes.Enumerable(context.Compilation) is INamedTypeSymbol enumerableType)
+            {
+                var operationActionsContext = new UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.OperationActionsContext(
+                    context.Compilation,
+                    enumerableType);
+
+                var operationActionsHandler = context.Compilation.Language == LanguageNames.CSharp
+                    ? (OperationActionsHandler)new CSharpOperationActionsHandler(operationActionsContext)
```

Comment:
We do not recommend language specific handlers in the shared layer. Would it be possible to define an abstract handler which have language specific implementations in language specific projects?

Reply:
That's what it is. I was just too lazy to put it on the language-specific project. I'll change it.